arch/arm64/src/fvp-v8r/fvp_lowputc.S

```diff
@@ -50,24 +50,6 @@
  * Public Functions
  ****************************************************************************/
 
-/* PL011 UART initialization
- * xb: register which contains the UART base address
- * c: scratch register number
- */
-
-GTEXT(up_earlyserialinit)
-SECTION_FUNC(text, up_earlyserialinit)
```

Comment:
I do not fully understand why this is removed and not renamed. Could you please make a short explanation here?

Reply:
https://github.com/apache/nuttx/pull/8716 already add arm64_earlyserialinit but forget to remove up_earlyserialinit.

Reply:
But the things done here are completely different from what is done in `arm64_earlyserialinit`. This code is branched very early from ASM code and uses
```
GTEXT(arm64_lowputc)
SECTION_FUNC(text, arm64_lowputc)
    ldr   x15, =CONFIG_UART0_BASE
    early_uart_ready x15, w2
    early_uart_transmit x15, w0
    ret
```
for the low level output. If we remove the `up_earlyserialinit` then non of the lines like `PRINT(primary_boot, "- Ready to Boot Primary CPU\r\n")` will be ever printed.
I will take a look into https://github.com/apache/nuttx/pull/8716 , but seems like we might be breaking an important path of debugging early startup process

Reply:
I see many mess in the code like in `arm_nommuhead.S` that calls `showprogress` -> `arm_lowputc` before
```
#ifdef USE_EARLYSERIALINIT
	bl		arm_earlyserialinit
#endif
```
And that is fine if underlying BL did all the work to mux the pins and setup the UART registers, but most probably will be unless in other case.
Some of the MCUs have `xxx_lowsetup()` that is similar to what `up_earlyserialinit` is doing here.
I think that we should hold off a bit here as this is not just removing of something unused and re-inspect how the things should work

Reply:
@qinwei2004 will provide the patch to correct the early print support after this get merged.

Reply:
@acassis @jerpelea @pkarashchenko 
it need to do early initialize when we want to print something in arm64_head.S
I cannot call arm64_earlyserialinit or other C function because it's very early (before BSS initialize)
and with a high ELs
At next patch, I will add arm64_earlyprintinit which is only initialize serial just setting BAUDRATE to bring serial into base work state in assembly language. A complete initiation will be done at arm64_earlyserialinit

I consider someone can also logging the booting message through rewriting arm64_lowputc with other debug method (eg semihosting, ARM debug channel etc), not only serial

but I check the function again and happy to find we don't need initialize it anymore. new version qemu or new option has do it for us. so I remove this code

Reply:
But in `bl arm64_earlyserialinit` you are doing exactly what you say you can't do "I cannot call arm64_earlyserialinit or other C function because".